### DISCUSSION TITLE
Link to View or Action

### DISCUSSION THREAD

**Discussion by nhereveri** [gonzalezreal/swift-markdown-ui#236] [Q&A]
Could you implement a functionality that allows InlineLink (or some other new element) to execute actions instead of just linking to URLs (such as opening another specific view or modifying a state)?

The idea is not to use a custom scheme and then use NavigationLink because I need to create them within Markdown. For example, when creating a single paragraph with two links to different views (as happened with a client who needed the terms and conditions in the same text along with the privacy policies).

**Comment by gonzalezreal** [comment]
Hi @nhereveri,

SwiftUI provides that functionality out of the box. You can implement a custom `OpenURLAction` and apply it to a view hierarchy's environment:

```swift
Markdown {
  """
  Visit the [terms](https://www.example.com) and [conditions](https://apple.com).
  """
}
.environment(\.openURL, OpenURLAction { url in
  // Modify a state, execute an action, etc., based on 'url'
  return .handled
})
```

For example, the companion Demo application uses this technique to scroll to the appropriate heading in the view when the user taps on an anchored link instead of opening the link in the browser.

https://github.com/gonzalezreal/swift-markdown-ui/blob/dba08f8641638b4310bea87eaf9f27ffedffd398/Examples/Demo/Demo/RepositoryReadmeView.swift#L126
